app/services/scrape.rb

```diff
@@ -1,117 +1,237 @@
       end
     rescue => e
-      puts "#{url}...#{e}"
+      logger.info "Test Id : #{test_id} Url: #{url.url} Error: #{e}"
     end
   end
 
+  def self.cms_and_version(html)
+    is_wordpress(html) || is_drupal(html) || is_shopify(html) || is_joomla(html)
+  end
+  
+  def self.is_wordpress(html)
+    wordpress_and_version_hash = check_wordpress_in_meta(html) || Hash.new
+    if !wordpress_and_version_hash.present? and check_wordpress_in_html(html)
+      wordpress_and_version_hash[:cms] = "wordpress"
+      version_from_resource = find_wordpress_version(html)
+      wordpress_and_version_hash[:cms_version] = version_from_resource
+    end
+    return wordpress_and_version_hash 
+  end
+
+  def self.is_drupal(html)
+    # some code here
+    return nil
+  end
+
+  def self.is_shopify(html)
+    # some code here
+    return nil
+  end
+
+  def self.is_joomla(html)
+    # some code here
+    return nil
+  end
   def self.check_wordpress_in_meta(html)
-    metaName = ['generator', 'Generator']
-    metaName.each do |name|
+    meta_name = ['generator', 'Generator']
+    meta_name.each do |name|
       html.search("meta[name='#{name}']").map do |line|
         if line['content']
           cms = line['content']
-          version = check_wordpress_name(cms)
-          return version if version
+          wordpress_and_version_hash = check_wordpress_name(cms)
+          return wordpress_and_version_hash if wordpress_and_version_hash
         end
       end
     end
-    return false
+    return nil
   end
 
   def self.check_wordpress_name(cms)
-    if cms && cms['Wordpress'] || cms['wordpress'] || cms['WordPress']
-      wordpressAndVersion = cms.split(' ')
-      return wordpressAndVersion[1] ? wordpressAndVersion[1] : "version not found"
+    if cms && ( cms['Wordpress'] || cms['wordpress'] || cms['WordPress'] )
+      wordpress_and_version_arr = cms.split(' ')
+      return {:cms => "wordpress", :cms_version => wordpress_and_version_arr[1]}
+    end
+    return nil
+  end
+
+  def self.check_wordpress_in_html(html)
+    if html.inner_text.match(/wp-content/)
+      return true
+    end
+  end
+
+  def self.find_wordpress_version(html)
+    return find_version_in_resource(Tags::LINK, html) || find_version_in_resource(Tags::SCRIPT, html)
+  end
+
+  def self.find_version_in_resource(resource, html)
+    lines = html.css(resource)
+    lines.each do |line|
+      version = find_version_in_sub_resource(line, SubSource::SRC) || find_version_in_sub_resource(line, SubSource::HREF)
+      return version if version 
     end
-    return nil;
+    return nil
+  end
+
+  def self.find_version_in_sub_resource(line, sub_resource)
+    checks = ['wp-includes/css/dist/block-library/style.min.css', 'wp-includes/js/wp-embed.min.js']
+    if line[sub_resource]
+      checks.each do |check|
+        if line[sub_resource][check]
+          version = line[sub_resource].split('ver=')[1]
```

Comment:
This doesn't give you wordpress version, rather it gives the version of the file. but for now, wordpress releases are in sync with these file versions

Not a reliable check